src/BlockTypes/ProductButton.php

```diff
@@ -64,6 +97,30 @@ protected function render( $attributes, $content, $block ) {
 					)
 				)
 			);
+
+			wc_store(
+				array(
+					'state' => array(
+						'woocommerce' => array(
+							'inTheCartText' => sprintf(
+								/* translators: %s: product number. */
+								__( '%s in the cart', 'woo-gutenberg-products-block' ),
+								'###'
+							),
+						),
+					),
+				)
+			);
+
+			$context = array(
+				'woocommerce' => array(
+					'isLoading'     => false,
+					'numberOfItems' => $number_of_items_in_cart,
+					'productId'     => $product->get_id(),
+					'addToCartText' => null !== $product->add_to_cart_text() ? $product->add_to_cart_text() : __( 'Add to cart', 'woo-gutenberg-products-block' ),
```

Comment:
It is necessary to put the `addToCartText` into the $context since a developer can update the text via `woocommerce_product_add_to_cart_text`.

Reply:
Yes, potentially yes.

Reply:
Ok, thanks 🙂